docs/content/en/docs/troubleshooting/troubleshooting.md

```diff
@@ -205,6 +197,105 @@ do
   kubectl get $crd -A
 done
 ```
+
+### Upgrade command stuck on `Waiting for external etcd to be ready`
+
+There can be a few reasons if the upgrade command is stuck on `waiting for external etcd for workload cluster to be ready` and eventually times out.
+
+First, check the underlying infrastructure to see if any new etcd machines are created.
+
+#### No new etcd machines are created in the infrastrure provider
+
+If no etcd machines are created, check the `etcdadm-bootstrap-provider-controller-manager`, `etcdadm-controller-controller-manager`, `capi-controller-manager`, `capi-kubeadm-control-plane-controller-manager`, and other Cluster API controller logs using the commands mentioned in [Generic cluster unavailable]({{< relref "#generic-cluster-unavailable" >}}) section.
+
+#### No IP assigned to the new etcd machine
+
+Refer to [no IP assigned to a VM]({{< relref "#no-ip-assigned-to-a-vm" >}}) section.
+
+### Machines are unhealthy after restoring CAPI control from the backup
+
+When an EKS Anywhere management cluster loses its CAPI resources, you need use the backup stored in the <cluster-name> folder to regain the management control. It is possible that after recovering the cluster from CAPI backup, the machine objects are in unhealthy/not ready state. For example:
+
+```sh
+$ kubectl get machines -n eksa-system
+
+NAME                                    PHASE
+cluster1-etcd-2md6m                     Failed
+cluster1-etcd-s8qs5                     Failed
+cluster1-etcd-vqs6w                     Failed
+cluster1-md-0-75584b4fccxfgs86-pfk22    Failed
+cluster1-x6rzc                          Failed
+```
+
+This can happen when the new machines got rolled out after the backup was taken, thus the backup objects being applied do not have the latest machine configurations. When applying an outdated CAPI backup, the provider machine object is out of sync with the actual infrastructure VM.
+
+
+#### etcdadm join failed in etcd machine bootstrap log
+
+When an etcd machine object is not in ready state, you can ssh into the etcd node and check the etcd bootstrap log:
+
+{{< tabpane >}}
+{{< tab header="Ubuntu or RHEL" lang="bash" >}}
+cat /var/log/cloud-init.log
+{{< /tab >}}
+{{< tab header="Bottlerocket" lang="bash" >}}
+sudo sheltie
+journalctl _COMM=host-ctr
+{{< /tab >}}
+{{< /tabpane >}}
+
+If the bootstrap log indicates that the etcdam join operation fail, this can mean that the etcd cluster contains unhealthy memeber(s). Looking at the log details you can further troubleshoot from below sections.
```

Comment:
```suggestion
If the bootstrap log indicates that the etcdadm join operation fail, this can mean that the etcd cluster contains unhealthy member(s). Looking at the log details you can further troubleshoot from below sections.
```